### PR TITLE
Add bitrix24 service

### PR DESCRIPTION
**Pull request by ruscon**

### CODE REVIEW

File: examples/bitrix24.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Example of retrieving an authentication token of the Github service
```
Comment (elliotchance) [inline examples/bitrix24.php]: Fix this text.

**Comment by elliotchance** [comment]
@ruscon please pull in the latest master to rerun the build.

File: src/OAuth/OAuth2/Service/Bitrix24.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    const SCOPE_CALENDAR = 'calendar';
+    const SCOPE_USER = 'user';
+    const SCOPE_ENTITY = 'entity';
+    const SCOPE_TASK = 'task';
+    const SCOPE_TASKS_EXTENDED = 'tasks_extended';
+    const SCOPE_IM = 'im';
+    const SCOPE_LOG = 'log';
+    const SCOPE_SONET_GROUP = 'sonet_group';
+
+    public function __construct(
```
Comment (elliotchance) [inline src/OAuth/OAuth2/Service/Bitrix24.php]: You can remove this constructor.

**Comment by ruscon** [comment]
done

File: src/OAuth/OAuth2/Service/Bitrix24.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            'client_secret' => $this->credentials->getConsumerSecret(),
+            'redirect_uri'  => $this->credentials->getCallbackUrl(),
+            'grant_type'    => 'authorization_code',
+            'scope'         => $this->scopes
+        );
+
+        $parameters['scope'] = implode(' ', $this->scopes);
+
+        // Build the url
+        $url = clone $this->getAccessTokenEndpoint();
```
Comment (elliotchance) [inline src/OAuth/OAuth2/Service/Bitrix24.php]: You don't need to clone the object as `getAccessTokenEndpoint()` returns a new new object.

Comment (elliotchance) [inline src/OAuth/OAuth2/Service/Bitrix24.php]: After this last change, this looks good to go :+1:

**Comment by Lusitanian** [comment]
thanks!!

**Comment by elliotchance** [comment]
https://github.com/Lusitanian/PHPoAuthLib/releases/tag/v0.5.0

**Pull request merged**